src/EventStore.Core/TransactionLog/Chunks/TFChunkDb.cs

```diff
@@ -22,13 +22,20 @@ public class TFChunkDb : IAsyncDisposable {
 	private readonly ITransactionFileTracker _tracker;
 	private int _closed;
 
-	public TFChunkDb(TFChunkDbConfig config, ITransactionFileTracker tracker = null, ILogger log = null, DbTransformManager transformManager = null) {
+	public TFChunkDb(
+		TFChunkDbConfig config,
+		ITransactionFileTracker tracker = null,
+		ILogger log = null,
+		DbTransformManager transformManager = null,
+		Action<EventStore.Core.Data.ChunkInfo> onChunkLoaded = null,
+		Action<EventStore.Core.Data.ChunkInfo> onChunkCompleted = null,
+		Action<EventStore.Core.Data.ChunkInfo> onChunkSwitched = null) {
```

Comment:
It's better for readability to move these optional callbacks to init-only property. `NoOp` delegate is not necessary because null-safe invocation can be done with `callback?.Invoke` expression.

Reply:
if we move them to init-only properties in TFChunkDb, we'll also need to change `TFChunkManager Manager` to an init-only property as we build the TFChunkManager in TFChunkDb's constructor (this will affect many tests, the TFChunkDb class is used in ~99 places)

but for the TFChunkManager class, moving them to init-only properties should be fine

Reply:
Valid concern